src/main/scala/rps/RPSMoves.scala

```diff
@@ -9,28 +10,18 @@ a little bit too much for the average use-case.
 The "agile-version" of enumerations are called
 "sealed trait extended by case objects".
 
-Here we define a RPSMoves ADT that will contain
-useful definitions and methods to craft RPS Moves.
-
+The buildo versions of the enumerations are called:
+```enumero```
 */
 
-sealed abstract class RPSMove(val value : String)
-object RPSMoves {
-
-    case object Rock extends RPSMove("0")
-    case object Paper extends RPSMove("1")
-    case object Scissors extends RPSMove("2")
-
-    def show(m: RPSMove) = m match {
-        case Rock => "💎"
-        case Paper => "📄"
-        case Scissors => "✂️"
-    }
-
-    // I need to find a way to obtain this set automagically
-    val setOfMoves: Set[RPSMove] = Set(Rock, Paper, Scissors)
-
-    def factoryMove(aMoveString: String): Option[RPSMove] = {
-        setOfMoves.filter(_.value == aMoveString).headOption
-    }
+/* 
+I will use the @indexedEnum annotation, that
+builds enumerations that follow the library's 
+convention for ADT-based enums with an associated value.
+*/
+@indexedEnum trait RPSMoves {
```

Comment:
ultra-minor, but usually we use the singular form for the name.

Reply:
Will be changed in the next revision!